### ISSUE TITLE
host.vars and group.vars are now empty

### ISSUE THREAD

**Issue by willthames** [ansible/ansible#16777]
##### ISSUE TYPE
- Bug Report
##### ANSIBLE VERSION

```
ansible 2.2.0 (devel 372018dfce) last updated 2016/07/21 16:07:03 (GMT +1000)
  lib/ansible/modules/core: (detached HEAD 7de287237f) last updated 2016/07/20 13:59:30 (GMT +1000)
  lib/ansible/modules/extras: (detached HEAD 68ca157f3b) last updated 2016/07/20 13:59:30 (GMT +1000)
  config file = 
  configured module search path = Default w/o overrides
```
##### CONFIGURATION

N/A
##### OS / ENVIRONMENT

N/A
##### SUMMARY

Since @jimi-c's change in 0ba9a6a87, ansible-inventory-grapher can no longer determine where in inventory a variable comes from. 

This is because after that change, `host.vars` (where `host` is an `inventory.Host` object) seems to be empty, whereas before, it contained the host's variables. Similarly, `group.vars` no longer contains the group's variables. 

I don't mind coding around this change but I don't know where to obtain just the variables for a host, and just the variables for a specific group
##### STEPS TO REPRODUCE

Use the following as inv_test.py in a checked clone of ansible-inventory-grapher: 

```
import sys

from ansible.inventory import Inventory
from ansible.parsing.dataloader import DataLoader
from ansible.vars import VariableManager

variable_manager = VariableManager()
loader = DataLoader()
inventory = Inventory(loader=loader, variable_manager=variable_manager, host_list=sys.argv[1])
variable_manager.set_inventory(inventory)
hosts = inventory.list_hosts(sys.argv[2])
print hosts[0].vars
groups = hosts[0].get_groups()
print [g for g in groups if g.name !='all'][0].vars
```
##### EXPECTED RESULTS

```
$ cd ~/src/ansible; git checkout 0ba9a6a87^
$ cd src/ansible-inventory-grapher
$ python inv_test.py test/inventory host
{u'gp_overridden_in_child': u'child', u'child_only': u'child', u'parent_overridden_in_child': u'child'}
{u'gp_overridden_in_parent': u'gp', u'gp_overridden_in_child': u'gp', u'gp_not_overridden': u'gp'}
```
##### ACTUAL RESULTS

```
$ cd ~/src/ansible; git checkout 0ba9a6a87
$ cd ~/src/ansible-inventory-grapher
$ python inv_test.py test/inventory host
{}
{}
```

**Comment by jctanner** [comment]
Related PR https://github.com/ansible/ansible/pull/16779

**Comment by willthames** [comment]
If for some reason this is essential for 2.1.1, I'll just use http://stackoverflow.com/questions/196960/can-you-list-the-keyword-arguments-a-python-function-receives to detect whether or not to pass `return_results`

**Comment by jimi-c** [comment]
# Closing This Ticket

Fixed by merge of #16779, any calls to the modified methods just needs to set the `return_results` flag to `True` now.

If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular issue is resolved.

Thank you!
